File: 0222-count-complete-tree-nodes/0222-count-complete-tree-nodes.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func countNodes(root *TreeNode) int {
    var answer int
	var dfs func(*TreeNode)
	dfs = func(node *TreeNode) {
		if node == nil {
			return
		}
		answer++
		dfs(node.Left)
		dfs(node.Right)
	}
    dfs(root)
	return answer
}

File: 0222-count-complete-tree-nodes/0222-count-complete-tree-nodes.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int countNodes(TreeNode root) {
        int[] answer = {0};
        dfs(root, answer);
        return answer[0];
    }

    public void dfs(TreeNode node, int[] count) {
        if (node == null) return;
        count[0]++;
        dfs(node.left, count);
        dfs(node.right, count);
    }
}